Detectors/ITSMFT/ITS/tracking/cuda/src/ClusterLinesGPU.cu

```diff
@@ -0,0 +1,287 @@
+                  determinantFirst +
+                (secondLine.cosinesDirector[1] * secondLine.cosinesDirector[1] * covarianceSecond[0] +
+                 secondLine.cosinesDirector[0] * secondLine.cosinesDirector[0] * covarianceSecond[1]) /
+                  determinantSecond;
+
+  mBMatrix[0] =
+    (firstLine.cosinesDirector[1] * covarianceFirst[2] * (-firstLine.cosinesDirector[1] * firstLine.originPoint[0] + firstLine.cosinesDirector[0] * firstLine.originPoint[1]) +
+     firstLine.cosinesDirector[2] * covarianceFirst[1] * (-firstLine.cosinesDirector[2] * firstLine.originPoint[0] + firstLine.cosinesDirector[0] * firstLine.originPoint[2])) /
+    determinantFirst;
+
+  mBMatrix[0] +=
+    (secondLine.cosinesDirector[1] * covarianceSecond[2] * (-secondLine.cosinesDirector[1] * secondLine.originPoint[0] + secondLine.cosinesDirector[0] * secondLine.originPoint[1]) +
+     secondLine.cosinesDirector[2] * covarianceSecond[1] *
+       (-secondLine.cosinesDirector[2] * secondLine.originPoint[0] +
+        secondLine.cosinesDirector[0] * secondLine.originPoint[2])) /
+    determinantSecond;
+
+  mBMatrix[1] =
+    (firstLine.cosinesDirector[0] * covarianceFirst[2] * (-firstLine.cosinesDirector[0] * firstLine.originPoint[1] + firstLine.cosinesDirector[1] * firstLine.originPoint[0]) +
+     firstLine.cosinesDirector[2] * covarianceFirst[0] * (-firstLine.cosinesDirector[2] * firstLine.originPoint[1] + firstLine.cosinesDirector[1] * firstLine.originPoint[2])) /
+    determinantFirst;
+
+  mBMatrix[1] +=
+    (secondLine.cosinesDirector[0] * covarianceSecond[2] * (-secondLine.cosinesDirector[0] * secondLine.originPoint[1] + secondLine.cosinesDirector[1] * secondLine.originPoint[0]) +
+     secondLine.cosinesDirector[2] * covarianceSecond[0] *
+       (-secondLine.cosinesDirector[2] * secondLine.originPoint[1] +
+        secondLine.cosinesDirector[1] * secondLine.originPoint[2])) /
+    determinantSecond;
+
+  mBMatrix[2] =
+    (firstLine.cosinesDirector[0] * covarianceFirst[1] * (-firstLine.cosinesDirector[0] * firstLine.originPoint[2] + firstLine.cosinesDirector[2] * firstLine.originPoint[0]) +
+     firstLine.cosinesDirector[1] * covarianceFirst[0] * (-firstLine.cosinesDirector[1] * firstLine.originPoint[2] + firstLine.cosinesDirector[2] * firstLine.originPoint[1])) /
+    determinantFirst;
+
+  mBMatrix[2] +=
+    (secondLine.cosinesDirector[0] * covarianceSecond[1] * (-secondLine.cosinesDirector[0] * secondLine.originPoint[2] + secondLine.cosinesDirector[2] * secondLine.originPoint[0]) +
+     secondLine.cosinesDirector[1] * covarianceSecond[0] *
+       (-secondLine.cosinesDirector[1] * secondLine.originPoint[2] +
+        secondLine.cosinesDirector[2] * secondLine.originPoint[1])) /
+    determinantSecond;
+
+  computeClusterCentroid();
+}
+
+GPUd() ClusterLinesGPU::ClusterLinesGPU(const int& firstLabel, const Line& firstLine, const int& secondLabel, const Line& secondLine,
```

Comment:
what is the meaning of these labels which are never used? Also, passing the reference on ``const int`` costs more than passing by value.

Reply:
To be fair this constructor was going to be eventually removed, it was a temptative porting of the corresponding ClusterLines present on CPU.
In the end I didn't end up using the one with the labels, there was a rationale behind, but the rest had not be implemented because I moved to the histogram approach.
Next commits I will just remove it, together with some other little artifacts. 

About the const ref being more costly I just did not know. I'll change it perhaps looking in the rest of the code. Does it happen only with trivial types like `ints` or its a general rule? I thought it was good practice to declare const wherever it possible and passing by reference wherever it possible.
 
Thanks for the advises.

Reply:
Ok, I am right now pruning a lot of the unused stuff (unused datamemeber included) in this class.